tests/unit/test_results.py

```diff
@@ -0,0 +1,170 @@
+import copy
+import datetime
+import random
+
+import responses
+
+from intezer_sdk.analyses_history import AnalysesHistory
+from intezer_sdk.api import get_global_api
+from intezer_sdk.results import Results
+from tests.unit.base_test import BaseTest
+
+BASE_DATA = AnalysesHistory._data_analyses_history(
+    start_date=datetime.datetime.now() - datetime.timedelta(days=3),
+    end_date=datetime.datetime.now()
+)
+
+NO_RESULTS = {'total_count': 0, 'analyses': []}
+NORMAL_RESULT = {
+    'total_count': 2,
+    'analyses': [{'account_id': '123'}, {'account_id': '456'}]
+}
+
+FILE_ANALYSES_REQUEST = '/analyses/history'
+URL_ANALYSES_REQUEST = '/url-analyses/history'
+ENDPOINT_ANALYSES_REQUEST = '/endpoint-analyses/history'
+
+
+class ResultsSpec(BaseTest):
+    def test_fetch_page_no_result(self):
```

Comment:
let's rename the tests to this convention: `test_fetch_page_raises_stop_iteration_when_no_more_pages_left`